src/cli/self_update/windows.rs

```diff
@@ -505,103 +501,83 @@ fn _apply_new_path(new_path: Option<Vec<u16>>) -> Result<()> {
 // Get the windows PATH variable out of the registry as a String. If
 // this returns None then the PATH variable is not a string and we
 // should not mess with it.
-fn get_windows_path_var() -> Result<Option<Vec<u16>>> {
-    use std::io;
+fn get_windows_path_var() -> Result<Option<String>> {
+    let environment = CURRENT_USER.create("Environment")?;
 
-    let root = RegKey::predef(HKEY_CURRENT_USER);
-    let environment = root
-        .open_subkey_with_flags("Environment", KEY_READ | KEY_WRITE)
-        .context("Failed opening Environment key")?;
-
-    let reg_value = environment.get_raw_value("PATH");
-    match reg_value {
-        Ok(val) => {
-            if let Some(s) = from_winreg_value(&val) {
-                Ok(Some(s))
-            } else {
-                warn!(
-                    "the registry key HKEY_CURRENT_USER\\Environment\\PATH is not a string. \
-                       Not modifying the PATH variable"
-                );
-                Ok(None)
-            }
+    match environment.get_string("PATH") {
+        Ok(val) => Ok(Some(val)),
+        Err(ref e) if e.code() == HRESULT::from_win32(ERROR_FILE_NOT_FOUND) => {
```

Comment:
I'm wondering if this code still makes sense? It seems surprising that getting the `PATH` from the registry would yield an `ERROR_FILE_NOT_FOUND` error...

Reply:
I have the same question, but that fact is that I must use this code or tests will failed with hint "File not found".

Reply:
Okay, that's enough for me to be satisfied that we still need it!

Reply:
@djc It actually can, see #3779 for such a case.